### DISCUSSION TITLE
Passing parameters into job

### DISCUSSION THREAD

**Discussion by andy-bout** [quartznet/quartznet#2101] [Q&A]
I have been reviewing the documentation and creating test schedules but cannot figure out how to pass parameters into a job.  Specifically need to pass in a process id (have a table that stores all the process we need to run and the associated metrics, errors, etc. for the process that is run is stored in the associated process tables).  Cannot figure out how to do that using quartz.  We will be using a SQL Server data store.  Having the ability to create the jobs from Quartz database entries and allowing the parameters to be set in the database  seems to be the ideal. Pretty sure I am missing something obvious and would appreciate any guidance.

**Comment by lahma** [comment]
Maybe this sample helps: https://github.com/quartznet/quartznet/blob/3.x/src/Quartz.Examples/04_JobParametersAndJobsStateMaintenance/

You can pass parameters both via job and trigger configuration, they will be available via MergedJobDataMap.

Never hurts to read the documentation either: https://www.quartz-scheduler.net/

**Comment by andy-bout** [comment]
Hi Marko,

Thank you very much.  The example helped a lot and I managed to get it to accept the parameters.  I did try to read as much of the documentation as I could and tried various methods.  I am not a programmer by education and deal mostly from a business design perspective but currently trying to investigate various methodologies for a very large financial business line of application that we are building so hence the lack of technical programming knowledge.  Trying my hardest to understand both quartznet and hangfire (as well as Azure and other scheduling tools) to determine if we can use an existing toolset or if we need to build our own scheduling framework).

Again, thank you and I will continue to read the documentation (it is easier to understand the documentation when seeing the examples and I hadn’t seen those examples before you sent them to me).

Andy

Andy Bout
Managing Partner | TruPoint Advisory
p: +1 403 607 2915
w: +1 587 997 1090  ext 700
e: ***@***.******@***.***>
***@***.***<https://trupointadvisory.com/>
________________________________
